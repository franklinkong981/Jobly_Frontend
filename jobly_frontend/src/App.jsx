import React from "react";
import './App.css';
import {BrowserRouter, Routes, Route} from "react-router-dom";

import JoblyNavbar from "./components/JoblyNavbar.jsx";
import Homepage from "./components/home/Homepage.jsx";
import LoginForm from "./components/auth/LoginForm.jsx";
import SignupForm from "./components/auth/SignupForm.jsx";
import UpdateProfileForm from "./components/profile/UpdateProfileForm.jsx";
import CompanyList from "./components/companies/CompanyList.jsx";
import CompanyDetail from "./components/companies/CompanyDetail.jsx";
import JobList from "./components/jobs/JobList.jsx";

function App() {

  return (
    <div className="App">
      <BrowserRouter>
        <JoblyNavbar/>
        <Routes>
          <Route path="/login" element={<LoginForm />} />
          <Route path="/signup" element={<SignupForm />} />
          <Route path="/profile" element={<UpdateProfileForm />} />
          <Route path="/companies" element={<CompanyList />} />
          <Route path="/companies/:name" element={<CompanyDetail />} />
          <Route path="/jobs" element={<JobList />} />
          <Route path="/" element={<Homepage />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
