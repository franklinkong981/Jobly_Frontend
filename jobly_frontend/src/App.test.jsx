import React from "react";
import {render} from "@testing-library/react";
import {MemoryRouter} from 'react-router-dom';

import App from "./App";

it("renders the entire Jobly app without crashing", function() {
  render((
    <MemoryRouter initialEntries={['/']} initialIndex={0}>
      <App />
    </MemoryRouter>
  ))
});