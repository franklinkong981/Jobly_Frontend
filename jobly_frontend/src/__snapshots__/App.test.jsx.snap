// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="JoblyNavbar navbar navbar-expand-md"
    >
      <a
        class="navbar-brand"
        data-discover="true"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item mr-4"
        >
          <a
            class="nav-link"
            data-discover="true"
            href="/login"
          >
            Log In
          </a>
        </li>
        <li
          class="nav-item mr-4"
        >
          <a
            class="nav-link"
            data-discover="true"
            href="/signup"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="Routes pt-5"
    >
      <div
        class="Homepage"
      >
        <div
          class="Homepage-logged-out container text-center"
        >
          <h1
            class="Homepage-logged-out-title mb-4 font-weight-bold"
          >
            Welcome to Jobly!
          </h1>
          <p
            class="Homepage-logged-out-subtitle lead"
          >
            Find all companies and job openings here in one convenient place.
          </p>
          <div
            class="Homepage-logged-out-links"
          >
            <a
              class="Homepage-logged-out-login-link btn btn-secondary font-weight-bold mr-3"
              data-discover="true"
              href="/login"
            >
              Log In
            </a>
            <a
              class="Homepage-logged-out-signup-link btn btn-info font-weight-bold"
              data-discover="true"
              href="/signup"
            >
              Create Account
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
